src/main/java/com/redhat/management/approval/Logger.java

```diff
@@ -0,0 +1,269 @@
+    writerThread.interrupt();
+  }
+
+  public void debug(String msg) {
+    log("debug", msg);
+  }
+
+  public void info(String msg) {
+    log("info", msg);
+  }
+
+  public void warn(String msg) {
+    log("warning", msg);
+  }
+
+  public void error(String msg) {
+    log("err", msg);
+  }
+
+  private void log(String level, String msg) {
+    eventQueue.add(createEvent(level, msg));
+    synchronized (eventQueue) {
+      eventQueue.notify();
+    }
+  }
+
+  private InputLogEvent createEvent(String level, String msg) {
+    Instant now = Instant.now();
+    TreeMap<String, Object> hMsg = new TreeMap<String, Object>();
+    hMsg.put("@timestamp", now.toString());
+    hMsg.put("hostname", hostName);
+    hMsg.put("pid", Logger.PID);
+    hMsg.put("tid", String.valueOf(Thread.currentThread().getId()));
+    hMsg.put("level", level);
+    hMsg.put("message", msg);
+    if (httpRequestId.get() != null) {
+      hMsg.put("request_id", httpRequestId.get());
+    }
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    String json = null;
+    try {
+      json = objectMapper.writeValueAsString(hMsg);
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
+      json = msg;
+    }
+
+    InputLogEvent event = new InputLogEvent();
+    event.setMessage(json);
+    event.setTimestamp(now.toEpochMilli());
+
+    return event;
+  }
+
+  private static String getProcessId() {
+    RuntimeMXBean runtimeBean = ManagementFactory.getRuntimeMXBean();
+    String jvmName = runtimeBean.getName();
+    return jvmName.split("@")[0];
+  }
+
+  public void run() {
+    boolean nonstop = true;
+    while(nonstop) {
+      synchronized (eventQueue) {
+        try {
+          eventQueue.wait();
+        } catch(InterruptedException e) {
+          // should not matter; wait again
+        }
+      }
+
+      // reschedule by canceling the old one and creating a new one
+      if (futureHandle != null && !futureHandle.isDone()) {
+        futureHandle.cancel(false);
+      }
+      futureHandle = this.scheduler.schedule(futureTask, FORCE_FLUSH_INTERVAL, TimeUnit.MILLISECONDS);
+
+      sendLogs(false);
+    }
+  }
+
+  public synchronized void sendLogs(boolean force) {
+    long now = System.currentTimeMillis();
+
+    if(eventQueue.isEmpty()) {
+      lastFlushTime = now;
+      return;
+    }
+
+    if (!force && now - lastFlushTime < FORCE_FLUSH_INTERVAL) {
+      return;
+    }
+
+    List<InputLogEvent> logs = new ArrayList<InputLogEvent>();
+    eventQueue.drainTo(logs);
+
+    for(InputLogEvent log : logs) {
+      System.out.println(log.getMessage());
+    }
+
+    if (client != null) {
+      logToCloud(logs);
+    }
+    lastFlushTime = now;
+  }
+
+  public void setHttpRequestId(String id) {
+    httpRequestId.set(id);
+  }
+
+  private AWSLogsClient initCloudWatchClient() {
+    if (this.accessKeyId == null || this.secretKey == null || this.logGroupName == null || this.logStreamName == null) {
+      return null;
```

Comment:
Should we output a message to System.out for this case? Otherwise there will be no clue to know why cloud watch has no logs